application/forms/ChannelForm.php

```diff
@@ -229,35 +156,162 @@ protected function onSuccess()
             return;
         }
 
-        $channel = [
-            'name' => $this->getValue('name'),
-            'type' => $this->getValue('type')
+        $channel = $this->getValues();
+        $channel['config'] = json_encode($this->getValue('config'));
+        if ($this->channelId === null) {
+            $this->db->insert('channel', $channel);
+        } else {
+            $this->db->update('channel', $channel, ['id = ?' => $this->channelId]);
+        }
+    }
+
+    /**
+     * Create config elements for the given channel type
+     *
+     * @param string $type The channel type
+     */
+    protected function createConfigElements(string $type): void
+    {
+        /** @var string $configs */
+        $configs = AvailableChannelType::on($this->db)
+            ->filter(Filter::equal('type', $type))
+            ->columns('config_attrs')
+            ->first()
+            ->config_attrs;
+
+        /** @var array<int, stdClass> $elementsConfig */
+        $elementsConfig = json_decode($configs, false);
+        var_dump($elementsConfig);
+        if (empty($elementsConfig)) {
+            return;
+        }
+
+        /** @var FormElementDecorator $decorator */
+        $decorator = $this->getDefaultElementDecorator();
+        $config = (new FieldsetElement('config'))
+            ->setDefaultElementDecorator($decorator);
```

Comment:
I thought fieldsets propagate that automatically? Why is this required?

Reply:
The fieldset object must first be added as an element so that all children can use the decorator.